integration/network_test.go

```diff
@@ -0,0 +1,861 @@
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/aoscloud/aos_servicemanager/config"
+	"github.com/aoscloud/aos_servicemanager/networkmanager"
+)
+
+/***********************************************************************************************************************
+ * Types
+ **********************************************************************************************************************/
+
+type trafficData struct {
+	lastUpdate   time.Time
+	currentValue uint64
+}
+
+type testTrafficStorage struct {
+	chains map[string]trafficData
+}
+
+/***********************************************************************************************************************
+ * Vars
+ **********************************************************************************************************************/
+
+var tmpDir string
+
+/***********************************************************************************************************************
+ * Init
+ **********************************************************************************************************************/
+
+func init() {
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp: false,
+		TimestampFormat:  "2006-01-02 15:04:05.000",
+		FullTimestamp:    true,
+	})
+	log.SetLevel(log.DebugLevel)
+	log.SetOutput(os.Stdout)
+}
+
+/***********************************************************************************************************************
+ * Main
+ **********************************************************************************************************************/
+
+func TestMain(m *testing.M) {
+	var err error
+
+	if tmpDir, err = ioutil.TempDir("", "aos_"); err != nil {
+		log.Fatalf("Error creating tmp folder: %v", err)
+	}
+
+	firewall, err := iptables.New()
+	if err != nil {
+		log.Fatalf("Can't create iptables instance: %v", err)
+	}
+
+	exists, err := checkPolicyForwardAcceptAll(firewall)
+	if err != nil {
+		log.Fatalf("Can't check policy exists: %v", err)
+	}
+
+	if exists {
+		if err := firewall.ChangePolicy("filter", "FORWARD", "DROP"); err != nil {
+			log.Fatalf("Can't create iptables rule: %v", err)
+		}
+	}
+
+	ret := m.Run()
+
+	if err := os.RemoveAll(tmpDir); err != nil {
+		log.Errorf("Can't remove tmp folder: %v", err)
+	}
+
+	if exists {
+		if err := firewall.ChangePolicy("filter", "FORWARD", "ACCEPT"); err != nil {
+			log.Errorf("Can't create iptables rule: %v", err)
+		}
+	}
+
+	os.Exit(ret)
+}
+
+func TestAddNetwork(t *testing.T) {
+	cases := []struct {
+		countInstance int
+		nameInstance  string
+	}{
+		{
+			countInstance: 10,
+			nameInstance:  "instance1",
+		},
+		{
+			countInstance: 15,
+			nameInstance:  "instance2",
+		},
+	}
+
+	manager, err := networkmanager.New(&config.Config{WorkingDir: tmpDir}, nil)
+	if err != nil {
+		t.Fatalf("Can't create network manager: %v", err)
+	}
+	defer manager.Close()
+
+	for numberNet, tCases := range cases {
```

Comment:
tCases -> testCase or what tCases stands for?

Reply:
Done

Reply:
use testCase, i just saw in some opensource projects that they use the name tCases for the table test implementation